### ISSUE TITLE
Finish swagger documentation.

### ISSUE THREAD

**Issue by AlvaroIsrael** [AlvaroIsrael/gobarber-app#153]
Swagger api documentation must be finished.

**Comment by AnujaChangede** [comment]
> Swagger api documentation must be finished.

Hi @AlvaroIsrael! Are you still looking for help for API Documentation? I would be happy to help.

**Comment by AlvaroIsrael** [comment]
Sure @AnujaChangede, fell free to help. Sorry for the late reply, I have been really busy these last few months, but i have not completed the swagger documentation yet.

**Comment by zahertalab** [comment]
Where to find your document to finish it?

**Comment by AlvaroIsrael** [comment]
Hey @zahertalab, the swagger documentation is hosted here: https://backend.gobarberapp.net/api/v1/docs/
The routes can be found here: https://github.com/AlvaroIsrael/gobarber-app/blob/master/gobarber-backend/src/shared/infra/http/routes/index.ts
To be able to find the payloads you would have to navigate inside the controllers of each missing documented api route and edit the swagger file located here: https://github.com/AlvaroIsrael/gobarber-app/blob/master/gobarber-backend/swagger.yaml
It is not going to be an easy job, but if you are willing to participate i can help you with!

**Comment by zahertalab** [comment]
> To be able to find the payloads you would have to navigate inside the controllers of each missing documented api route and edit the swagger file located here: https://github.com/AlvaroIsrael/gobarber-app/blob/master/gobarber-backend/swagger.yaml It is not going to be an easy job, but if you are willing to participate i can help you with!

Thank you. You mean by "payloads" parameters? and "route" endpoint? 

Just I should edit inside swagger file located in you respiratory instead of editing in swagger editor?

**Comment by AlvaroIsrael** [comment]
Yeah, you could use the online swagger editor, if you feel more comfortable with it, and them paste the result in the swagger.yaml, (some minor adjustments may be necessary tough), it will reflect changes in `/api/v1/docs` endpoint. In localhost it would be `http://localhost:3333/api/v1/docs` for example, so you can test if the copying and pasting worked!

**Comment by zahertalab** [comment]
Yes, thank you for trust. Is these any files or folders should I navigate find the payloads? Where to find the controllers?

**Comment by AlvaroIsrael** [comment]
Yes there is. The project is structured as:

src
  - modules
    - entity name (users, for example)
      - infra
        - http
          - controllers

You could start by looking at the ProfileController.ts which is one of the endpoints that need to be documented.
The 3 main entities we have in this small project are: appointments, notifications and users.

<img width="419" alt="Captura de Tela 2022-10-09 às 00 35 32" src="https://user-images.githubusercontent.com/8402007/194736835-02b66a4f-088a-42a4-8852-258ead419c94.png">

**Comment by zahertalab** [comment]
I'm a technical writer and I wan not a developer. I'm trying to break to API documentation. I have studied some API documentation courses. Do you think there is a way to understand this project requirements?

I supposes if there is an auto generated open API docs, I can enrich it. Or to instruct me to generate them?

Thanks,

**Comment by EleekaN** [comment]
Hi, anyone working on this? Can I take this up?

**Comment by AlvaroIsrael** [comment]
@EleekaN Hi, sure, you can help!

**Comment by AlvaroIsrael** [comment]
@zahertalab this complicate things a bit, if you don't have the programmings skills to understand the code, the only way i can think about, is if i was able to assist you. But i don't currently have time for that. I have been busy with work schedule and even my replies to this tread are late. Maybe you can try to find another project for you to contribute.

**Comment by EleekaN** [comment]
> Hi, anyone working on this? Can I take this up?



> @EleekaN Hi, sure, you can help!

Thank you, I will go through the details and update you soon.
